runtime/src/main/cpp/TestSupport.hpp

```diff
@@ -15,4 +21,46 @@ constexpr int kDefaultThreadCount = 10;
 constexpr int kDefaultThreadCount = 100;
 #endif
 
+// Performs minimal initialization of the current thread memory subsystem.
+// This initilization is enough for running tests for the C++ part for the stdlib.
+//  - For the new MM: registeres the current thread in the thread registry.
+//  - For the legacy MM: does nothing, returns nullptr.
+MemoryState* InitMemoryForTests();
+
+// Deinitiliazes memory subsystem used for C++ stdlib tests.
+//  - For the new MM: unregisteres the current thread, nullify current thread data.
+//  - For the legacy MM: does nothing.
+void DeinitMemoryForTests(MemoryState* state);
+
+// Scopely initializes the runtime for tests.
+class ScopedRuntimeInit : private kotlin::Pinned {
+public:
+    ScopedRuntimeInit() : memoryState_(InitMemoryForTests()) {
+        test_support::InitRuntimeForTests(memoryState());
```

Comment:
If we move `GetMemoryState` into each mm, could we use real `InitMemory` and `DeinitMemory` instead?

Reply:
Yes. My original intent was to make this "initialization for tests" as minimal as possible. But it seems that using the regular `InitMemory` will be more clear. Fixed.